cmd/generator/main.go

```diff
@@ -246,6 +255,57 @@ func getReleasePlugin(ctx context.Context, client *github.Client, repositoryName
 	return plugin, nil
 }
 
+func downloadSignatures(assets []github.ReleaseAsset) ([]*model.PluginSignatures, error) {
+	signatures := make([]*model.PluginSignatures, 0, len(assets))
+	for _, asset := range assets {
+		hash, err := getHashFromSignatureFileName(asset)
+		if err != nil {
+			return nil, err
+		}
+		sig, err := getSignatureFromSignatureFileName(asset)
+		if err != nil {
+			return nil, err
+		}
+
+		signature := &model.PluginSignatures{
+			PublicKeyHash: hash,
+			Signature:     sig,
+		}
+		signatures = append(signatures, signature)
+	}
+	return signatures, nil
+}
+
+func getHashFromSignatureFileName(asset github.ReleaseAsset) (string, error) {
+	name := asset.GetName()
+	if !strings.HasSuffix(name, ".sig") && !strings.HasSuffix(name, ".asc") {
+		return "", errors.New("signature file has wrong extension")
+	}
+	name = name[:len(name)-4]
```

Comment:
Perhaps [strings.TrimSuffix](https://golang.org/pkg/strings/#TrimSuffix) instead?

Reply:
I have to check the suffix with `HasSuffix` and after that call the `TrimSuffix` which will once again call the `HasSuffix` and after that return `name[:len(name)-len(suffix)]`. I did not like that . 
Probably better idea just to add a comment that we are trimming the suffix?

Reply:
Personally, `HasSuffix` and `TrimSuffix` are much more clear than a comment and `name[:len(name)-4]`. I think we should optimize for readability over theoretical performance here.